apps/css-workshop/src/components/Badge.astro

```diff
@@ -0,0 +1,32 @@
+---
+type Props = {
+  backgroundColor?:
+    | 'skyblue'
+    | 'celery'
+    | 'froly'
+    | 'steelblue'
+    | 'sunglow'
+    | 'seabuckthorn'
+    | 'montecarlo'
+    | 'poloblue'
+    | 'bouquet'
+    | 'ash'
+    | 'oak';
+  status?: 'informational' | 'positive' | 'warning' | 'negative';
+} & astroHTML.JSX.HTMLAttributes;
+
+const { class: className, backgroundColor, status, ...props } = Astro.props;
```

Comment:
- Astro provides all externally passed as `Astro.props` ([See docs](https://docs.astro.build/en/basics/astro-components/#component-props)). We can define a type/interface for it using `type Props`, but it's not strictly necessary, because these components are for internal use. I only added the `type` to demonstrate it.
- The `{}` syntax is used for ["destructuring"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) specific properties from `Astro.props`. The `...props` is a catch-all syntax for any props that we don't want to manipulate directly.
- `class: className` is basically renaming the variable to `className` when destructuring, because `class` is a reserved JavaScript keyword.